src/tribler/core/components/database/db/layers/tests/test_user_activity_layer.py

```diff
@@ -0,0 +1,201 @@
+from typing import Generator
+
+from pony.orm import db_session
+from pytest import fixture
+
+from tribler.core.components.database.db.layers.user_activity_layer import UserActivityLayer
+from tribler.core.components.user_activity.types import InfoHash
+from tribler.core.utilities.pony_utils import TrackedDatabase
+
+
+@fixture(name="layer")
+def fixture_activity_layer() -> Generator[UserActivityLayer, None, None]:
+    database = TrackedDatabase()
+    database.bind(provider="sqlite", filename=":memory:")
+    ual = UserActivityLayer(database)
+    database.generate_mapping(create_tables=True)
+    yield ual
+    database.disconnect()
+
+
+def float_equals(a: float, b: float) -> bool:
+    return round(a, 5) == round(b, 5)
+
+
+def test_store_no_losers(layer: UserActivityLayer) -> None:
+    """
+    Test that queries can be stored and retrieved.
+    """
+    layer.store("test query", InfoHash(b"\x00" * 20), set())
+
+    with db_session():
+        queries = layer.Query.select()[:]
+
+        assert len(queries) == 1
+        assert queries[0].query == "test query"
+        assert len(queries[0].infohashes) == 1
+        assert list(queries[0].infohashes)[0].infohash == b"\x00" * 20
+        assert float_equals(list(queries[0].infohashes)[0].preference, 1.0)
```

Comment:
NIT Same logic, but a bit more correct and easier to read.

The suggested version uses `Query.get()` instead of `Query.select()` because there should only be a single entity in the database, which you assert later. It also confines the `with` block to only include lines that actually need the `db_session`. The suggested version avoids multiple, yet identical, conversions to a list and retrieving the first element of the list.

```suggestion
    with db_session():
        test_query = layer.Query.get()
        infohashes = list(test_query.infohashes)

    assert test_query.query == "test query"
    assert len(infohashes) == 1
    
    infohash = infohashes.pop()
    assert infohash.infohash == b"\x00" * 20
    assert float_equals(infohash.preference, 1.0)
```